enhancements/api-review/custom-tags-aws.md

```diff
@@ -31,113 +36,300 @@ have the ability to apply user defined tags to many resources created by OpenShi
+
+### Limitations
+
+- User-defined tags for PV volumes cannot be updated but only set during creation  of volume using `--extra-tags`.Present CSI spec does not support `UpdateVolume` to update volume metadata.
+   Restarting csi driver to initiate multiple `CreateVolume` will depend on underlying storage layer idempotency implementation for the operation. It is recommended to use a interface from CSI spec which enforces idempotency guarantees for volume metadata updates.
+   In case of AWS EC2  `CreateVolume`, returns `IdempotentParameterMismatch` error when identical request is not found for the same client token.
+   Also, the CSI spec `https://github.com/container-storage-interface/spec/blob/master/spec.md` mentions grpc error code
+   `6 ALREADY_EXISTS` for `a volume corresponding to the specified volume name already exists but is incompatible with the specified capacity_range, volume_capabilities, parameters, accessibility_requirements or volume_content_source`.
+- User-defined tags cannot be updated for vpc, security groups, elb, route53, subnet resources as there is no operator managing the resources post installation.
+   User-defined tags cannot be updated to an AWS resource which is not managed by an operator in openshift-* namespace.
+- User-defined tags can be updated on the following AWS resources.
+  1. EC2 instances for master and worker nodes.
+  2. Image registry.
+  3. Ingress LB.
+  4. IAM credentials by CCO in mint mode of operation.
 
 ## Proposal
 
-New `experimentalPropagateUserTags` field added to `.platform.aws` of install config to indicate that the user tags should be applied to AWS
-resources created by in-cluster operators.
+- Existing `experimentalPropagateUserTags` will be renamed to `propagateUserTags` in `.platform.aws` of install config to indicate that the user tags should be applied to AWS
+resources created by in-cluster operators.\
+  `experimentalPropagateUserTags` field will be set for deprecation.
+
+  If `propagateUserTags` is set to true, install validation will fail if there is any tag that starts with `kubernetes.io` or `openshift.io`.
+
+- Add a new field `resourceTags` to `.spec.platformSpec.aws` of the `Infrastructure.config.openshift.io` type. Tags included in the
+  `resourceTags` field will be applied to new resources created for the cluster. The `resourceTags` field will be populated by the installer only if the `propagateUserTags` field is true.
+
+  `.spec.platformSpec.aws` is a mutable field and `.status.platformStatus.aws` is immutable.
+  `.status.platformStatus.aws` will have older version tags defined and is required for upgrade case.
+
+- All operators that create AWS resources (ingress, cloud credential, storage, image registry, machine-api) will apply these tags to all AWS resources they create.\
+  Operator must update the AWS resource within (5 minutes + cloud provide API response time).\
+  Operator must update only when there is a change in `.spec.platformSpec.aws.resourceTags`.\
```

Comment:
Yes, correct. The statement was written to convey an implementation note for the case, "if there is no update in tags, do not overwrite the tags on AWS resource". As this might end up in too many calls to AWS. Cool, rewriting it.